prow/cmd/deck/main.go

```diff
@@ -1663,3 +1670,40 @@ func defaultLensRemoteConfig(lfc *config.LensFileConfig) error {
 
 	return nil
 }
+
+// ValidatePath validates various parts (e.g. bucket, folder, etc.) of a storage path.
+func ValidatePath(cfg config.Getter, path string) error {
+	if !cfg().Deck.EnableWhitelist {
+		return nil
+	}
+
+	parts := strings.SplitN(path, "/", 4)
+	if len(parts) < 4 {
+		return fmt.Errorf("invalid path: %s expected <key-type>/<bucket>/<folder>/<sub-folder...>", path)
+	}
+
+	// Check bucket
+	bucket := parts[1]
+	bwl := sets.String{}
```

Comment:
I think this and fwl below can be a bit clearer.

Reply:
I cleaned the set insertion steps up a bit.